ConsoleApp1/ConsoleApp1/Program.cs

```diff
@@ -0,0 +1,528 @@
+                int x = ar[0].x;
+                int y = ar[0].y;
+                Points = ar.ToList();
+                Points.RemoveAt(0);
+                Matrix.Screen[x, y] = ".";
+            }
+
+            public static bool Scan(Rotate r, out bool eat, out int ax, out int ay)
+            {
+                var ar = Points.ToArray();
+                int x = ar[ar.Length - 1].x;
+                int y = ar[ar.Length - 1].y;
+                ax = x;
+                ay = y;
+                Points = ar.ToList();
+                try
+                {
+                    switch (r)
+                    {
+                        case Rotate.Up:
+                            if (Matrix.Screen[x - 1, y] == "." || Matrix.Screen[x - 1, y] == "O")
+                            {
+                                if (Matrix.Screen[x - 1, y] == "O")
+                                {
+                                    eat = true; RemoveApple(x - 1, y);
+                                }
+                                else eat = false;
+                                return true;
+                            }
+                            else
+                            {
+                                eat = false;
+                                return false;
+                            }
+                            break;
+                        case Rotate.Down:
+                            if (Matrix.Screen[x + 1, y] == "." || Matrix.Screen[x + 1, y] == "O")
+                            {
+                                if (Matrix.Screen[x + 1, y] == "O")
+                                {
+                                    eat = true; RemoveApple(x + 1, y);
+                                }
+                                else eat = false;
+                                return true;
+                            }
+                            else
+                            {
+                                eat = false;
+                                return false;
+                            }
+                            break;
+                        case Rotate.Left:
+                            if (Matrix.Screen[x, y - 1] == "." || Matrix.Screen[x, y - 1] == "O")
+                            {
+                                if (Matrix.Screen[x, y - 1] == "O")
+                                {
+                                    eat = true; RemoveApple(x, y - 1);
+                                }
+                                else eat = false;
+                                return true;
+                            }
+                            else
+                            {
+                                eat = false;
+                                return false;
+                            }
+                            break;
+                        case Rotate.Right:
+                            if (Matrix.Screen[x, y + 1] == "." || Matrix.Screen[x, y + 1] == "O")
```

Comment:
each case is almost copypaste of previous, I'm sure it can be extracted to single method

Reply:
It can be. But method will do the same. With same quantity of variants, or in 2 methods "scan for move" and "scan for apples". But i don's see reason.